include/material/bingham.tcc

```diff
@@ -75,14 +75,17 @@ Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
   // D_1^2 + D_2^2 + 2*D_3^2 + 2*D_4^2 + 2*D_5^2 Yielding is defined: rate of
   // shear > critical_shear_rate_^2 Checking yielding from strain rate vs
   // critical yielding shear rate
-  const double shear_rate_squared =
-      2 * (strain_rate.dot(strain_rate) +
-           strain_rate.tail(3).dot(strain_rate.tail(3)));
+  double shear_rate =
+      std::sqrt(2. * (strain_rate.dot(strain_rate) +
+                     strain_rate.tail(3).dot(strain_rate.tail(3))));
 
   // Apparent_viscosity maps shear rate to shear stress
-  double apparent_viscosity = 0;
-  if (shear_rate_squared > critical_shear_rate_ * critical_shear_rate_)
-    apparent_viscosity = 2 * ((tau0_ / (std::sqrt(shear_rate_squared))) + mu_);
+  // Check if shear rate is 0
+  double apparent_viscosity = 0.;
+  if (shear_rate * shear_rate > critical_shear_rate_ * critical_shear_rate_) {
+    if (shear_rate < shear_rate_threshold) shear_rate = shear_rate_threshold;
```

Comment:
I'm not convinced with this check. Is this when you set `critical_shear_rate_` to be zero? Even setting `shear_rate` to a small value like 1.e-15 would induce a very high viscosity!

Reply:
It's for both. `critical_shear_rate_` needs to be at certain value.
I know it's a high viscosity, but `shear_rate = 0` is even worse right?

Reply:
`shear_rate` can't be zero inside the condition where `shear_rate^2` > critical_shear_rate^2. No reason to do the next check if it is for `0`

Reply:
Understood. I agree. I removed it but I would like to rename the variable to the one in this pull request